pkg/collector/corechecks/gpu/nvml.go

```diff
@@ -0,0 +1,84 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2024-present Datadog, Inc.
+
+//go:build linux
+
+// Package gpu defines the agent corecheck for
+// the GPU integration
+package gpu
+
+import (
+	"errors"
+	"fmt"
+	"sync"
+
+	"github.com/NVIDIA/go-nvml/pkg/nvml"
+)
+
+var initOnce sync.Once
+
+type gpuDevice struct {
+	nvml.Device
+}
+
+func wrapNvmlError(ret nvml.Return) error {
+	if ret == nvml.SUCCESS {
+		return nil
+	}
+
+	return errors.New(nvml.ErrorString(ret))
+}
+
+func ensureNvmlInit() error {
```

Comment:
why not use `init()` instead? we need one initialization per process (as this library will be used by system-probe and agent processes eventually)

Reply:
In that case we'd also try to enable it even when it's not going to be used, no?

Reply:
you are right, resolving this comment